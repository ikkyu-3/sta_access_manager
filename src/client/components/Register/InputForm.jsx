// @flow
import type { RegisterInputFormProps } from "@types";

import * as React from "react";
import Input from "@components/Register/Input";
import SingleButton from "@components/SingleButton";
import styles from "./style.css";

class InputForm extends React.Component<RegisterInputFormProps> {
  componentDidMount() {
    this.props.selectedMenu();
    this.props.onChangeStep();
  }

  componentWillUnmount() {
    this.props.unmount();
  }

  render() {
    const {
      canShow,
      firstNameValue,
      firstNameError,
      onChangeFirstName,
      lastNameValue,
      lastNameError,
      onChangeLastName,
      mailAddressValue,
      mailAddressError,
      onChangeMailAddress,
      onClickNextButton,
      onSubmitForm
    } = this.props;
    const { main, inputName, inputEmail, domain, contentWithButton } = styles;
    const domainName = process.env.MAIL_DOMAIN;

    return (
      <div className={main}>
        <form onSubmit={onSubmitForm}>
          <div className={contentWithButton}>
            <div className={inputName}>
              <Input
                id="lastName"
                label="姓"
                error={lastNameError}
                value={lastNameValue}
                canShow={canShow}
                onChange={onChangeLastName}
              />
              <Input
                id="firstName"
                label="名"
                error={firstNameError}
                value={firstNameValue}
                canShow={canShow}
                onChange={onChangeFirstName}
              />
            </div>
            <div className={inputEmail}>
              <Input
                id="email"
                label="Email"
                error={mailAddressError}
                value={mailAddressValue}
                canShow={canShow}
                onChange={onChangeMailAddress}
              />
              <span className={domain}>@{domainName}</span>
            </div>
          </div>
          <SingleButton
            text="次へ"
            className="next"
            onButtonClick={onClickNextButton}
          />
        </form>
      </div>
    );
  }
}
export default InputForm;
