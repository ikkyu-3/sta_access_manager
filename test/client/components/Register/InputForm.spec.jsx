import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import InputForm from "@components/Register/InputForm";

const fn = () => {};
const domain = "exsample.com";
process.env.MAIL_DOMAIN = domain;

describe("Register/InputForm.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(
        <InputForm
          selectedMenu={fn}
          onChangeStep={fn}
          unmount={fn}
          canShow={false}
          firstNameValue="firstName"
          firstNameError=""
          onChangeFirstName={fn}
          lastNameValue="lastName"
          lastNameError=""
          onChangeLastName={fn}
          mailAddressValue="email"
          mailAddressError=""
          onChangeMailAddress={fn}
          onClickNextButton={fn}
          onSubmitForm={fn}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const selectedMenuMock = jest.fn();
    const onChangeStepMock = jest.fn();
    const unmount = jest.fn();

    let inputForm;

    beforeAll(() => {
      inputForm = shallow(
        <InputForm
          selectedMenu={selectedMenuMock}
          onChangeStep={onChangeStepMock}
          unmount={unmount}
          canShow={false}
          firstNameValue="firstName"
          firstNameError=""
          onChangeFirstName={fn}
          lastNameValue="lastName"
          lastNameError=""
          onChangeLastName={fn}
          mailAddressValue="email"
          mailAddressError=""
          onChangeMailAddress={fn}
          onClickNextButton={fn}
          onSubmitForm={fn}
        />
      );
    });

    it("componentDidMountは、selectedMenu, onChangeStepを実行する", () => {
      expect(selectedMenuMock).toBeCalled();
      expect(onChangeStepMock).toBeCalled();
    });

    it("componentWillUnmountは、unmountを実行する", () => {
      inputForm.instance().componentWillUnmount();
      expect(unmount).toBeCalled();
    });

    it("process.env.MAIL_DOMAINに指定した値が、メールアドレスのドメイン名に表示されている", () => {
      expect(inputForm.find(".domain").text()).toBe(`@${domain}`);
    });
  });
});
