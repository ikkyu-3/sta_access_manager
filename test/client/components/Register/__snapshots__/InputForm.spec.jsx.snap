// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register/InputForm.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="main"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="contentWithButton"
    >
      <div
        className="inputName"
      >
        <div
          className="inputArea"
        >
          <label
            htmlFor="lastName"
          >
            姓
            <span
              className="message hidden"
            >
              
            </span>
            <input
              className="input"
              id="lastName"
              onChange={[Function]}
              type="text"
              value="lastName"
            />
          </label>
        </div>
        <div
          className="inputArea"
        >
          <label
            htmlFor="firstName"
          >
            名
            <span
              className="message hidden"
            >
              
            </span>
            <input
              className="input"
              id="firstName"
              onChange={[Function]}
              type="text"
              value="firstName"
            />
          </label>
        </div>
      </div>
      <div
        className="inputEmail"
      >
        <div
          className="inputArea"
        >
          <label
            htmlFor="email"
          >
            Email
            <span
              className="message hidden"
            >
              
            </span>
            <input
              className="input"
              id="email"
              onChange={[Function]}
              type="text"
              value="email"
            />
          </label>
        </div>
        <span
          className="domain"
        >
          @
          exsample.com
        </span>
      </div>
    </div>
    <div
      className="singleButton"
    >
      <button
        className="next"
        onClick={[Function]}
      >
        次へ
      </button>
    </div>
  </form>
</div>
`;
